src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/CustomDatasetSource.kt

```diff
@@ -59,14 +60,25 @@ class CustomDatasetSource private constructor(
             .build()
     }
 
+    @Deprecated(message = "Use CustomDatasetSource.storeInS3(location, jira) instead",
+        replaceWith = ReplaceWith("store(location, null)")
+    )
     fun storeInS3(
         location: StorageLocation
     ) {
-        store(location)
+        store(location, null)
+    }
+
+    fun storeInS3(
+        location: StorageLocation,
+        jira: Jira
```

Comment:
I think jira should be passed in the constructor and not here. The operation so far accepted "where to store", which was enough as the interface for "DatasetSource". `jira` represents "what to do before store" and I don't think that will change between different `storeInS3` executions.

Is there some technical reason why we can't pass `jira` in the constructor (`Builder`)? 

Maybe we could even skip the `if (jira == null)` if we'd model the dependency as "pre store operation" or just something like "jira stopper", "jira stop handler", "jira hand break"

Reply:
Good idea. Pass it via builder.
But the `jirahome` version is a bug, doesn't work for DC, better delete it altogether. Only the new calls with `Jira` will work properly. We can signal that via API:
1. make `Jira` a required parameter in the `Builder`
2. deprecate the old builder constructor in favor of the new one